src/app/config/Sample_Excel_Word.ts

```diff
@@ -0,0 +1,176 @@
+const ProgressBar = require('progress');
+const log = require('single-line-log').stdout;
+const Spinner = require('cli-spinner').Spinner;
+const { spawn } = require('child_process');
+const { exec } = require('child_process');
+const { execSync } = require('child_process');
+const readline = require('readline');
+
+import * as os from 'os';
+
+
+shell.config.silent = true;
+
+async function exec_script_Excel_Mail(){
+  // shell.exec('code .');
+  console.log('Welcome to experience this Office add-in sample: Excel Mail merge Add-in project!');
+
+  return new Promise<boolean>((resolve, reject) => {
+    
+        let is_vscode_installed = false;
+
+        console.log('Welcome to experience this Office add-in sample!');
+
+        // Step 1: Get sample code
+        console.log('Step [1/3]: Getting sample code...');
+        let spinner = new Spinner('Processing.. %s');
+        spinner.setSpinnerString('|/-\\');
+        spinner.start();
+
+        shell.exec('git clone https://github.com/OfficeDev/Excel-Scenario-based-Add-in-Samples.git', {async:true}, (code, stdout, stderr) => {
+            shell.cd('./Excel-Scenario-based-Add-in-Samples/Mail-Merge-Sample-Add-in');
+            // shell.exec('git sparse-checkout set Mail-Merge-Sample-Add-in/', {async:true}, (code, stdout, stderr) => {
+
+            spinner.stop(true);
+            readline.clearLine(process.stdout, 0);
+            readline.cursorTo(process.stdout, 0);
+
+            // Step 2: Check if VSCode is installed
+            console.log('Step [1/3] completed!');
+            console.log('Step [2/3]: Checking if Visual Studio Code is installed...');
+            if (shell.which('code')) {
+                console.log('Visual Studio Code is installed on your machine. Would open in VSCode for exploring the code.');
+                is_vscode_installed = true;
+                shell.exec('code -g README.md:22');
+            } else {
+                console.log('Visual Studio Code is not installed on your machine.');
+                shell.exec('start Mail-Merge-Sample-Add-in');
+            }
+
+            console.log('Step [2/3] completed!');
+            
+            // Ask user if sample Add-in automatic launch is needed
+            let rl = readline.createInterface({
+                input: process.stdin,
+                output: process.stdout
+            });
+
+            let auto_launch_answer = false;
+            rl.question('Do you want to continue with some operations? (Y/N)\n', (answer) => {
+                if (answer.trim().toLowerCase() == 'y') {
+                  // Continue with the operations
+                  // Step 3: Provide user the command to side-load add-in directly 
+                    console.log('Step [3/3]: Automatically side-load add-in directly...');
+                    spinner.start();
+
+                    shell.cd('./Mail-Merge-Sample-Add-in');
+                    shell.exec('npm install', {async:true}, (code, stdout, stderr) => {
```

Comment:
This is shipping code.  Even if you want to experiment with something it would have to be in ship quality.  Note that we don't have the option to flight something and then just turn it off.

The generator doesn't distinguish between the list of options and what it generates . . . how are you not displaying it and yet using the generator.